lib/logstash/codecs/multiline.rb

```diff
@@ -216,6 +216,8 @@ def flush(&block)
   end
 
   def auto_flush
+    return if @last_seen_listener.nil?
```

Comment:
would it be worth introducing a `using_listeners?` method or similar?
```
def using_listeners?
  !@last_seen_listener.nil?
end
```

that would have the benefit of making it a little more self-documenting?